irodsbackend/static/irodsbackend/js/irodsbackend.js

```diff
@@ -247,18 +270,24 @@ $(document).ready(function() {
                     }
                     objLink += obj['name'] + '</a>';
 
-                    var colSpan = '1';
-                    var icon = 'mdi:file-document-outline';
-                    var toolTip = 'File';
+                    var colSpan = obj['type'] === 'coll' ? '1' : '4';
+                    var icon = obj['type'] === 'coll' ? 'mdi:folder-open' : 'mdi:file-document-outline';
+                    var toolTip = obj['type'] === 'coll' ? 'Collection' : 'File';
+
+                    var copyButton = '';
                     if (obj['type'] === 'coll') {
-                        colSpan = '4';
-                        icon = 'mdi:folder-open';
-                        toolTip = 'Collection';
+                        copyButton = '<button class="btn btn-secondary sodar-list-btn pull-right" ' +
+                            'id="sodar-irods-copy-btn-' + obj['name'] + '" ' +
+                            'title="Copy iRODS path into clipboard" data-tooltip="tooltip" ' +
+                            'data-placement="top" onclick="copy_path(\'' + obj['path'] + '\', \'sodar-irods-copy-btn-' + obj['name'] + '\')">' +
+                            '<i class="iconify" data-icon="mdi:clipboard-text-multiple"></i>' +
+                            '</button>';
                     }
-                    var iconHtml = '<i class="iconify mr-1" data-icon="'+ icon +'"' +
-                        'title="'+ toolTip +'"></i>';
 
-                    body += '<tr><td colspan="' + colSpan + '">' + iconHtml + objLink + '</td>';
+                    var iconHtml = '<i class="iconify mr-1" data-icon="' + icon + '"' +
+                        ' title="' + toolTip + '"></i>';
+                    body += '<tr><td colspan="' + colSpan + '">' + iconHtml + objLink + copyButton + '</td>';
```

Comment:
There are three issues with the current functionality:

- The copy button is missing for data objects
- Cells for collections are not drawn beyond the first column
- The headers no longer correspond to table rows (probably a colspan issue?)